internal/appsec/listener/httpsec/http.go

```diff
@@ -84,37 +87,44 @@ func NewWAFEventListener(handle *waf.Handle, actions emitter.Actions, addresses
 			}))
 		}
 
-		values := make(map[string]any, 7)
+		runData := waf.RunAddressData{
+			Persistent: make(map[string]any, 8),
+		}
 		for addr := range addresses {
 			switch addr {
 			case HTTPClientIPAddr:
 				if args.ClientIP.IsValid() {
-					values[HTTPClientIPAddr] = args.ClientIP.String()
+					runData.Persistent[HTTPClientIPAddr] = args.ClientIP.String()
 				}
 			case ServerRequestMethodAddr:
-				values[ServerRequestMethodAddr] = args.Method
+				runData.Persistent[ServerRequestMethodAddr] = args.Method
 			case ServerRequestRawURIAddr:
-				values[ServerRequestRawURIAddr] = args.RequestURI
+				runData.Persistent[ServerRequestRawURIAddr] = args.RequestURI
 			case ServerRequestHeadersNoCookiesAddr:
 				if headers := args.Headers; headers != nil {
-					values[ServerRequestHeadersNoCookiesAddr] = headers
+					runData.Persistent[ServerRequestHeadersNoCookiesAddr] = headers
 				}
 			case ServerRequestCookiesAddr:
 				if cookies := args.Cookies; cookies != nil {
-					values[ServerRequestCookiesAddr] = cookies
+					runData.Persistent[ServerRequestCookiesAddr] = cookies
 				}
 			case ServerRequestQueryAddr:
 				if query := args.Query; query != nil {
-					values[ServerRequestQueryAddr] = query
+					runData.Persistent[ServerRequestQueryAddr] = query
 				}
 			case ServerRequestPathParamsAddr:
 				if pathParams := args.PathParams; pathParams != nil {
-					values[ServerRequestPathParamsAddr] = pathParams
+					runData.Persistent[ServerRequestPathParamsAddr] = pathParams
 				}
 			}
 		}
 
-		wafResult := listener.RunWAF(wafCtx, waf.RunAddressData{Persistent: values}, timeout)
+		if extractSchemas {
+			runData.Persistent["waf.context.processor"] = map[string]any{"extract-schema": true}
+		}
+
+		wafResult := listener.RunWAF(wafCtx, runData, timeout)
+		listener.AddAPISecurityTags(op, wafResult.Derivatives)
```

Comment:
It would, although a gating check would save a few jumps incase no derivatives are generated, which is most cases. I added a gating check with `wafResult.HasDerivatives()` since this seems the most logical.